File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.js
function solution(num) {
    var answer = num;
    var count = 0;
    
    while(answer!==1){
        if(count == 500){
            return -1;
        }
        if(answer%2==0){
            answer = answer/2;
            count+=1;
        }else{
            answer = (answer*3) + 1;
            count+=1;
        }
    }
    return count;
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        long n = num;
        int count = 0;

        while (true) {
            if (n == 1) {
                return count;
            }

            if (count == 500) {
                return -1;
            }

            if (n % 2 == 0) {
                n /= 2;
            } else {
                n = n * 3 + 1;
            }

            count++;
        }
    }
}